### PR TITLE
Use filesystemwatcher instead of while true read loop

### PR DESCRIPTION
**Pull request by secondfry**
Lowers CPU usage significantly.
Theoretically may introduce problem on old Linux boxes with really old mono.
Probably should be squashed upon merge.

### CODE REVIEW

**Comment by Rnen** [comment]
Uuuuh, did you just remove the ability to custom color MA messages?

File: mutliadmin/MultiAdmin/OutputThread.cs [hunk 1]
```diff
@@ -28,362 +28,354 @@ public static ConsoleColor MapConsoleColor(string color, ConsoleColor def = Cons
...
-					if (Directory.Exists(dedicatedDir))
+					if (!File.Exists(file))
 					{
-						activeDir = Directory.GetFiles(dedicatedDir, "sl*.mapi", SearchOption.TopDirectoryOnly).OrderBy(f => f).ToArray<string>();
+						server.Write("Message printer warning: Could not " + command + " " + file + ". File does not exist!", ConsoleColor.Yellow);
+						server.Write("skipping");
+						break;
 					}
+
+					StreamReader sr = new StreamReader(file);
```
Comment (Rnen) [inline mutliadmin/MultiAdmin/OutputThread.cs]:
```suggestion
					using(StreamReader sr = new StreamReader(file))
					{
```

File: mutliadmin/MultiAdmin/OutputThread.cs [hunk 2]
```diff
@@ -28,362 +28,354 @@ public static ConsoleColor MapConsoleColor(string color, ConsoleColor def = Cons
...
+					stream = sr.ReadToEnd();
+					command = "close";
+					sr.Close();
```
Comment (Rnen) [inline mutliadmin/MultiAdmin/OutputThread.cs]:
```suggestion
					sr.Close();
					}
```

File: mutliadmin/MultiAdmin/OutputThread.cs (hunk 1 above)
Comment (Rnen) [reply mutliadmin/MultiAdmin/OutputThread.cs]: This will properly dispose the streamreader

Comment (secondfry) [reply mutliadmin/MultiAdmin/OutputThread.cs]: Using closure has the same effect as calling `Close()` in the end.

From MSDN
This implementation of Close calls the Dispose method passing a true value.

Comment (Rnen) [reply mutliadmin/MultiAdmin/OutputThread.cs]: You still need to close the streamreader using this method, but it properly **__disposes__** the streamreader

**Comment by MichalPetryka** [comment]
Can't be merged, i've added it in my fork and it blocks on server restart, i probably fixed it in main game

**Comment by secondfry** [comment]
You mean, there are changes in SL sources which make this obsolete? 'cas I'm using this for more than 45 days without any problems.

**Pull request merged**